interface/modules/custom_modules/oe-module-ehi-exporter/src/EhiExporter.php

```diff
@@ -0,0 +1,723 @@
+            // TODO: @adunsulag need to test and make sure we get eligibility_verification AND benefit_eligibility as part of our export here.
+            $parentTraversalTables = self::PARENT_FK_TABLES_TRAVERSAL;
+            return in_array($tableName, $parentTraversalTables);
+        }
+        return false;
+    }
+
+    private function shouldExportAdditionalAssets($tableName)
+    {
+        $additionalAssets = ['form_painmap'];
+        return in_array($tableName, $additionalAssets);
+    }
+    private function exportAdditionalAssets(\ZipArchive $zip, $tableName)
+    {
+        $additionalAssets = [
+            'form_painmap' => [
+                ['name' => 'images/painmap.png', 'path' => $GLOBALS['webserver_root'] . "/interface/forms/painmap/templates/painmap.png"]
+            ]
+        ];
+        $assets = $additionalAssets[$tableName] ?? [];
+        foreach ($assets as $assetsToExport) {
+            if (file_exists($assetsToExport['path'])) {
+                if (!$zip->addFile($assetsToExport['path'], $assetsToExport['name'])) {
+                    $this->logger->errorLogCaller("File exists but failed to export to zip", ['path' => $assetsToExport['path']]);
+                }
+            } else {
+                $this->logger->errorLogCaller("Failed to export additional asset as file is missing", ['path' => $assetsToExport['path']]);
+            }
+        }
+    }
+
+    private function writeCsvFile($jobTask, &$records, $tableName, $outputLocation)
+    {
+        $uuidDefinition = UuidRegistry::getUuidTableDefinitionForTable($tableName);
+        if (!empty($uuidDefinition)) {
+            $convertUuid = true;
+        } else {
+            $convertUuid = false;
+        }
+        $columns = QueryUtils::listTableFields($tableName);
+        // note I am intentionally avoiding php://temp/maxmemory here which would be more performant but runs a higher risk of files being
+        // left around on the hard disk which we do not want to do.  Memory is harder to read against but does run the risk of overloading the server
+        // if there isn't enough RAM or if the php ini max memory setting is too low.
+        $csvFile = fopen("php://memory", 'r+');
+        fputcsv($csvFile, $columns);
+        $recordCount = 0;
+        foreach ($records as $record) {
+            if ($convertUuid && !empty($record['uuid'])) {
+                $record['uuid'] = UuidRegistry::uuidToString($record['uuid']);
+            }
+            fputcsv($csvFile, $record);
+            $recordCount++;
+        }
+        rewind($csvFile);
+        $dataContents = stream_get_contents($csvFile);
+        // free up memory by closing the connection and run the garbage collector since these files could be potentially
+        // huge if there is a lot of patients represented
+        fclose($csvFile);
+        unset($csvFile);
+        $encryptedContents = $this->cryptoGen->encryptStandard($dataContents);
```

Comment:
when doing encryptStandard or decryptStandard to item on the drive then you want to use the keys in the database, which will look like this:
```
$encryptedContents = $this->cryptoGen->encryptStandard($dataContents, null, 'database');
```

Reply:
Should be resolved now.